pkg/metrics/timer/timer.go

```diff
@@ -0,0 +1,76 @@
+package timer
```

Comment:
so personally, i feel the goal of this package should be `allow installer codebase to track various stages/units of work` 
another useful api provided would be a `Summarize` function that can summarize all the stages.

Now to achieve that, we should allow the user (here is could an asset, or the cmd package) to start a specific stage and end a specific stage easily, but still allowing another user (it could be the same asset, or cmd package)
so whoever holds the pointer to the timer-tracker can summarize all the timers.

so I would prefer this package api to look like

```
// global tracker to reduce the _how to distribute the timer_ problem for now
StartTimer(key string)
StopTimer(key string)
Summarize(something to control the output location)

// also allow some people to create an explicit ones for other use.
NewTracker() Tracker
Tracker struct
(Tracker) StartTimer(key string)
(Tracker) StopTimer(key string)
(Tracker) Summarize(...)

```

Reply:
@abhinavdahiya this design sounds good to me. one of the things that @rna-afk and I have been discussing is how these timer values will be attached to the metrics created in the metrics package. 

Do you have an opinion on whether:

1. The "user" pushes the value to metrics package after reading from this tracker
2. The tracker itself pushes the value (based on key)
3. Metrics package pulls values from this API

Looking at the suggested API I think the first option, which sounds good to me. I have been trying to avoid option 3.

Reply:
> 1. The "user" pushes the value to metrics package after reading from this tracker

I like this option, it will allows us to write the glue code that will extract these from tracker and push to metrics. so both these are separate and don't depend on one another.

Reply:
was discussing with @rna-afk and I'm not sure I get why we need a global tracker at all? 

> // global tracker to reduce the _how to distribute the timer_ problem for now

I'm trying to think of a reason in case we eventually we made certain sections concurrent, but I'm not sure that use case would make sense either.

Reply:
> // also allow some people to create an explicit ones for other use.
NewTracker() Tracker
Tracker struct
(Tracker) StartTimer(key string)
(Tracker) StopTimer(key string)
(Tracker) Summarize(...)

in my original comment, tracker tracks multiple timers. so if you have access to a locally defined tracker for example, an asset creating one for tracking various stages internally, the asset can start stop timers.

but for us today, esp in cmd/ there isn't a great place to define a tracker that can use used by everyone to start and stop their timers. 

> // global tracker to reduce the _how to distribute the timer_ problem for now
StartTimer(key string)
StopTimer(key string)
Summarize(something to control the output location)

That's where a global tracker comes into play where we can import the package and use the global tracker to interact.

that was my thought. hopefully that provides more context. @patrickdillon @rna-afk

Reply:
With regards to the input parameter of the Summarize function, am I right to think that the input would be something in the lines of debug levels or is it something more?